plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -102,10 +103,24 @@ func (p *podAffinityPriorityMap) processTerm(term *api.PodAffinityTerm, podDefin
 	}
 }
 
-func (p *podAffinityPriorityMap) processTerms(terms []api.WeightedPodAffinityTerm, podDefiningAffinityTerm, podToCheck *api.Pod, fixedNode *api.Node, multiplier int) {
+func (p *podAffinityPriorityMap) processTerms(terms[] api.WeightedPodAffinityTerm, podDefiningAffinityTerm, podToCheck *api.Pod, fixedNode *api.Node, multiplier int, podlister algorithm.PodLister) {
 	for i := range terms {
 		term := &terms[i]
-		p.processTerm(&term.PodAffinityTerm, podDefiningAffinityTerm, podToCheck, fixedNode, float64(term.Weight*multiplier))
+
+
+		// A crude implementation of on-node pod label selector scoring.
+		selector,_ := unversioned.LabelSelectorAsSelector(term.PodAffinityTerm.LabelSelector)
+		pods ,_ := podlister.List(selector)
+		for _, mypod := range pods {
+			if mypod.Spec.NodeName == fixedNode.Name {
+				p.Lock()
+				p.counts[fixedNode.Name] += float64(term.Weight * multiplier)
```

Comment:
i think this should be a non multiplied, but iirc there was a reason for the multiplication here , i.e. just to spread the scores